server/src/controllers/employer/getAllEmployers.js

```diff
@@ -1,14 +1,60 @@
 const logger = require("pino")();
+const { Op } = require("sequelize");
 const Employer = require("../../models/employer.model");
 
 const getAllEmployersRequestHandler = async (req, res) => {
   try {
-    const employers = await Employer.findAll();
+    const page = req?.query?.page ? parseInt(req.query.page, 10) : 0;
+    const pageSize = req?.query?.page ? parseInt(req.query.pageSize, 10) : 10;
+    const offsetSize = page * pageSize;
+
+    const {
+      employerName,
+      phoneNumber,
+      startDateAdded,
+      endDateAdded,
+      ownerId,
+      postalCode,
+    } = req.query;
+
+    const query = {};
+    if (employerName) {
+      query.name = { [Op.like]: `%${employerName}%` };
+    }
+    if (phoneNumber) {
+      query.phone_number = { [Op.like]: `%${phoneNumber}%` };
+    }
+    if (startDateAdded) {
+      query.creation_date = { [Op.gte]: new Date(startDateAdded) };
+    }
+
+    if (endDateAdded) {
+      query.creation_date = {
+        ...query.creation_date,
+        [Op.lte]: new Date(endDateAdded),
+      };
+    }
+
+    if (ownerId && ownerId !== "-1") {
+      query.creator = ownerId;
```

Comment:
can you change from `query.creator = ownerId` to `query.owner = ownerId`. mb for misleading, I fixed in my pr too